test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -0,0 +1,489 @@
+		return u2s(arg[0]) < u2s(arg[1]) ? 1 : 0;
+	case Instruction::SGT:
+		return u2s(arg[0]) < u2s(arg[1]) ? 1 : 0;
+	case Instruction::EQ:
+		return arg[0] == arg[1] ? 1 : 0;
+	case Instruction::ISZERO:
+		return arg[0] == 0 ? 1 : 0;
+	case Instruction::AND:
+		return arg[0] & arg[1];
+	case Instruction::OR:
+		return arg[0] | arg[1];
+	case Instruction::XOR:
+		return arg[0] ^ arg[1];
+	case Instruction::BYTE:
+		return arg[0] >= 32 ? 0 : (arg[1] >> unsigned(8 * (31 - (arg[0] & 0xff)))) & 0xff;
+	case Instruction::SHL:
+		return arg[0] > 255 ? 0 : (arg[1] << unsigned(arg[0] & 0xff));
+	case Instruction::SHR:
+		return arg[0] > 255 ? 0 : (arg[1] >> unsigned(arg[0] & 0xff));
+	case Instruction::SAR:
+	{
+		static u256 const hibit = u256(1) << 255;
+		if (arg[0] >= 256)
+			return arg[1] & hibit ? u256(-1) : 0;
+		else
+		{
+			unsigned amount = unsigned(arg[0] & 0xff);
+			u256 v = arg[1] >> amount;
+			if (arg[1] & hibit)
+				v |= u256(-1) << (256 - amount);
+			return v;
+		}
+	}
+	case Instruction::ADDMOD:
+		return arg[2] == 0 ? 0 : u256((u512(arg[0]) + u512(arg[1])) % arg[2]);
+	case Instruction::MULMOD:
+		return arg[2] == 0 ? 0 : u256((u512(arg[0]) * u512(arg[1])) % arg[2]);
+	case Instruction::SIGNEXTEND:
+		if (arg[0] >= 31)
+			return arg[0];
+		else
+		{
+			unsigned testBit = unsigned(arg[0] & 0xff) * 8 + 7;
+			u256 ret = arg[1];
+			u256 mask = ((u256(1) << testBit) - 1);
+			if (boost::multiprecision::bit_test(ret, testBit))
+				ret |= ~mask;
+			else
+				ret &= mask;
+			return ret;
+		}
+	// --------------- blockchain stuff ---------------
+	case Instruction::KECCAK256:
+	{
+		if (!logMemoryRead(arg[0], arg[1]))
+			return u256("0x1234cafe1234cafe1234cafe") + arg[0];
+		uint64_t offset = uint64_t(arg[0] & uint64_t(-1));
+		uint64_t size = uint64_t(arg[1] & uint64_t(-1));
+		return u256(keccak256(bytesConstRef(m_state.memory.data() + offset, size)));
+	}
+	case Instruction::ADDRESS:
+		return m_state.address;
+	case Instruction::BALANCE:
+		return m_state.balance;
+	case Instruction::ORIGIN:
+		return m_state.origin;
+	case Instruction::CALLER:
+		return m_state.caller;
+	case Instruction::CALLVALUE:
+		return m_state.callvalue;
+	case Instruction::CALLDATALOAD:
+		return readZeroExtended(m_state.calldata, arg[0]);
+	case Instruction::CALLDATASIZE:
+		return m_state.calldata.size();
+	case Instruction::CALLDATACOPY:
+		if (logMemoryWrite(arg[0], arg[2]))
+			copyZeroExtended(
+				m_state.memory, m_state.calldata,
+				size_t(arg[0]), size_t(arg[1]), size_t(arg[2])
+			);
+		return 0;
+	case Instruction::CODESIZE:
+		return m_state.code.size();
+	case Instruction::CODECOPY:
+		if (logMemoryWrite(arg[0], arg[2]))
+			copyZeroExtended(
+				m_state.memory, m_state.code,
+				size_t(arg[0]), size_t(arg[1]), size_t(arg[2])
+			);
+		return 0;
+	case Instruction::GASPRICE:
+		return m_state.gasprice;
+	case Instruction::EXTCODESIZE:
+		logTrace(_instruction, arg);
+		return u256(keccak256(h256(arg[0]))) & 0xffffff;
```

Comment:
This doesn't look like ``EXTCODESIZE``... This is meant as a random placeholder? Same for the ones below?

Reply:
Yes, we have not the slightest idea about the outside world.

We want to guarantee that for `x=y`, we have `f(x) = f(y)` and if `x != y`, then `f(x)` is rather not `f(y)`, the rest is bonus.